src/hermes/commands/deposit/invenio.py

```diff
@@ -17,648 +17,613 @@
 import requests
 
 from hermes import config
+from hermes.commands.deposit.base import BaseDepositPlugin
 from hermes.commands.deposit.error import DepositionUnauthorizedError
 from hermes.error import MisconfigurationError
 from hermes.model.context import CodeMetaContext
 from hermes.model.path import ContextPath
 from hermes.utils import hermes_user_agent
 
-_DEFAULT_LICENSES_API_PATH = "api/licenses"
-_DEFAULT_COMMUNITIES_API_PATH = "api/communities"
-_DEFAULT_DEPOSITIONS_API_PATH = "api/deposit/depositions"
 
+_log = logging.getLogger("cli.deposit.invenio")
 
-def prepare(click_ctx: click.Context, ctx: CodeMetaContext):
-    """Prepare the deposition on an Invenio-based platform.
 
-    In this function we do the following:
+class InvenioClient(requests.Session):
```

Comment:
**praise:** In general, I like how you split up the functionality into Deposition and Client.